feedback-module/src/components/Module.js

```diff
@@ -110,23 +112,34 @@ function Module({ pagetitle, endpoint, dir }) {
       pageTitleAsScreen(currentPageTitle);
       pageChange(currentPageTitle, nextPageTitle);
 
-      sendFormData(screen.formID),
-        setScreen(SCREENS[nextScreen]),
-        setCheckboxError(false);
+      const submissionObj = requestInfo(screen.formID);
+      if (submissionObj) {
+        setLoading(true);
+        requestService(
+          screen.formID,
+          submissionObj,
+          () => {
+            setScreen(SCREENS[nextScreen]);
+            headerRef.current.scrollIntoView(true);
+          },
+          setFailedRequest,
+          setLoading
+        );
+        setFormData({});
+      }
     }
   };
 
   const handleSubmit = (e) => {
     e.preventDefault();
   };
 
-  const changeScreen = (text, nextScreen, feedbackID) => {
+  const changeScreen = (type, text, nextScreen, feedbackID) => {
     // If button contains a feedbackID, update the feedbackType of the feedback object
-    if (screen.formID) {
+    if (screen.formID && type === "submit") {
```

Comment:
Is this function ever called with a type other than "submit"?

Reply:
No, I don't think we'd want to submit data anytime other than when the button type is submit.

Reply:
So do we need to add this "type" parameter to the function and make this extra check?

Reply:
Sorry, I think I misunderstood your question— in order to collect data from multiple screens for one object (e.g. feedback data comes from both the first and second), I've added a `formID` property to the config for any screen that provides such data (before, we hardcoded that the first screen's data gets added to the feedback object). Since we don't want to trigger the submit on _all_ screens that have a `formID` (e.g. we don't want to trigger it to submit on the first screen), I've added this extra check to make sure that the button is actually a submit button.